### ISSUE TITLE
Passing _id in msg.projection not working

### ISSUE THREAD

**Issue by Gervwyk** [node-red/node-red#1144]
I have this very simple flow:
````
[{"id":"8db396d9.0fa948","type":"function","z":"8e99a9a2.4c3f28","name":"find","func":"\nvar newMsg = {};\nnewMsg.collection = 'chartsbar';\nnewMsg.operation  = 'find.toArray';\nnewMsg.payload    = {'company': 'MRM'};\nnewMsg.projection = {\"_id\": 1, \"title\": 1};\nreturn newMsg;","outputs":1,"noerr":0,"x":250,"y":2000,"wires":[["dd10281c.cef1c8"]]},{"id":"c1b781da.ad30d","type":"inject","z":"8e99a9a2.4c3f28","name":"","topic":"","payload":"","payloadType":"date","repeat":"","crontab":"","once":false,"x":120,"y":2000,"wires":[["8db396d9.0fa948"]]},{"id":"dd10281c.cef1c8","type":"debug","z":"8e99a9a2.4c3f28","name":"","active":true,"console":"false","complete":"true","x":390,"y":2000,"wires":[]}]
````
I'm trying to simply write a `newMsg.projection` property with a key of `_id: 1` but in my debug msg `_id` does not show. If I pass this to the `mongodb2` node to find documents and include the `_id` field, the field is not included and I cannot update the documents with matching `_id`. Any ideas? is there a way around this issue?

**Comment by knolleary** [comment]
Hi, `mongodb2` is not a node this project maintains. I suggest you contact its author - https://github.com/ozomer/node-red-contrib-mongodb2

**Comment by Gervwyk** [comment]
I agree but it is not related to `mongodb2`. It is related to the function node:
````
var newMsg = {};
newMsg.collection = 'chartsbar';
newMsg.operation  = 'find.toArray';
newMsg.payload    = {'company': 'MY'};
newMsg.projection = {"_id": 1, "title": 1};
return newMsg;
````
In the debug message from the function block I only see `{"title": 1}` for `msg.projection`, meaning that the `_id` field was not set in the function block.

**Comment by knolleary** [comment]
What version of node-red are you using? There was a bug in 0.16 where the debug sidebar was filtering properties that began with a `_` - which was fixed in 0.16.1 - but this was purely in what the debug sidebar displayed, the underlying message was unaffected.

I've just tested your function code on the latest release, 0.16.2, and the `projection` property has its `_id` property intact.

**Comment by Gervwyk** [comment]
Yes. I'm on 0.16.0. I'll update. thanks.
